### PR TITLE
chore: Build driver with vite

### PR DESCRIPTION
**Pull request by sainthkh**
- Closes 

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: patches/@cypress+unique-selector+0.4.4.patch [hunk 1]
```diff
@@ -0,0 +1,293 @@
+diff --git a/node_modules/@cypress/unique-selector/lib/index.esm.js b/node_modules/@cypress/unique-selector/lib/index.esm.js
+new file mode 100644
+index 0000000..60a443e
+--- /dev/null
```
Comment (lmiller1990) [inline patches/@cypress+unique-selector+0.4.4.patch:4]: Can we just fix an release a new cypress unique selector instead of patch package?

**Review by lmiller1990 — COMMENTED** [review]
Nice to see this!! I'm not sure how much is left but excited to see this, I'm guessing there's still a bit to do - I thought we had some kind of circular dependency problem in driver? 

I wonder why we need to polyfill `path` and `stream` and `util` - what's using that in driver?

File: patches/@cypress+unique-selector+0.4.4.patch (hunk 1 above)
Comment (sainthkh) [reply patches/@cypress+unique-selector+0.4.4.patch:4]: I just wanted to test if everything works with this change. I'll open a new PR at `unique-selector` later.

**Comment by sainthkh** [comment]
I checked the circular dependency problem with [`dpdm`](https://github.com/acrazing/dpdm). The result is that our circular dependency only exists for TypeScript types. So, we could simply move on.

I polyfill them because they were in `vite.config.mjs` of `driver`. I'll remove them and see what's going on.

File: patches/@cypress+unique-selector+0.4.4.patch (hunk 1 above)
Comment (sainthkh) [reply patches/@cypress+unique-selector+0.4.4.patch:4]: As you can see below, errors are flooding.

**Comment by lmiller1990** [comment]
I think it's fine to polyfill for now, just curious - it would be super nice to eventually move everything to ESM eventually and have a simple `vite.config` - our `webpack.config` is immensely complex right now, one of the nice things about Vite is you generally need less config.

**Comment by chrisbreiding** [comment]
@sainthkh Have you been able to make any progress on this PR?

**Comment by sainthkh** [comment]
@chrisbreiding 

I learned that Webpack generates some polyfills for global variables while Vite does not. And many packages have their own way of handling them, so it's a bit hard for me to finish this PR. Maybe researching the polyfill code of Webpack polyfill for `global`, `window`, `process` might help.

I'm sorry. I think I should close this for now.

**Pull request closed without merging**